python/ml4ir/base/io/file_io.py

```diff
@@ -45,6 +45,17 @@ def make_directory(dir_path: str, clear_dir: bool = False, log=None) -> str:
     return dir_path
 
 
+def read_list(infile: str):
+    """Read a text file"""
```

Comment:
It does more that just read, it converts it to a list + do manipulations. Could we be explicit in comments about the assumptions?